### DISCUSSION TITLE
copy 的文件，如何在 src 代码中知道生成出来的 hash 值

### DISCUSSION THREAD

**Discussion by crazyair** [umijs/umi#12053] [Q&A]
`copy:[{ from: a.js ,to: [name].[hash:8]}]`
业务中如何知道文件是
a.xxsfsfss

**Comment by fz6m** [comment]
你可以在构建结束后读一下产物目录里看看这个文件叫啥，如果想在 src 中就知道是不行的，你可以人工给 `a.js` 加版本号。

这里的 copy 功能是 [copy-webpack-plugin](https://github.com/webpack-contrib/copy-webpack-plugin) 做的，如果想知道具体算法，可以去看源码。

**Reply by crazyair** [reply]
查询了 umi 中 hash 代码找到 `createHash`，就改成如下了，不知道对不对，是不是推荐写法，这里得出 Icon 组件的 scriptUrl 的推荐写法后，我给 antd 文档修改下。

`scriptUrl` 是推荐用 `cdn` 链接，但是对于很多人还是有点成本的，所以还是要有一个本地 icon.js 的实现方式，还不像 umi 的 logo，icon.js 需要有 hash

https://github.com/umijs/umi/issues/12050#issuecomment-1890139581

**Reply by fz6m** [reply]
copy 这个功能是 [copy-webpack-plugin](https://github.com/webpack-contrib/copy-webpack-plugin) 这个 webpack 插件的功能，并不是 umi 实现的功能，所以你配置的 hash 是如何产生的，需要从 [copy-webpack-plugin](https://github.com/webpack-contrib/copy-webpack-plugin) 的源码寻找答案，不是 umi 里的 `createHash` 之类的代码。
